File: livingrimoire start here/LivinGrimoire java/src/Auxiliary_Modules/AXStringSplit.java
package Auxiliary_Modules;

public class AXStringSplit {
    // may be used to prepare data before saving or after loading
    // the advantage is less data fields. for example: {skills: s1_s2_s3}
    private String spChar = "_";

    public void setSpChar(String spChar) {
        this.spChar = spChar;
    }
    public String[] split(String s1){
        return s1.split(spChar);
    }
    public String stringBuilder(String[] sArr){
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < sArr.length - 1; i++) {
            sb.append(sArr[i]);
            sb.append(this.spChar);
        }
        sb.append(sArr[sArr.length - 1]);
        return sb.toString();
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/auxiliary_modules/AXStringSplit.kt
package auxiliary_modules

class AXStringSplit {
    // may be used to prepare data before saving or after loading
    // the advantage is less data fields. for example: {skills: s1_s2_s3}
    private var spChar = "_"
    fun setSpChar(spChar: String) {
        this.spChar = spChar
    }

    fun split(s1: String): Array<String> {
        return s1.split(spChar.toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
    }

    fun stringBuilder(sArr: Array<String?>): String {
        val sb = StringBuilder()
        for (i in 0 until sArr.size - 1) {
            sb.append(sArr[i])
            sb.append(spChar)
        }
        sb.append(sArr[sArr.size - 1])
        return sb.toString()
    }
}